### DISCUSSION TITLE
How to read an array form binary file?

### DISCUSSION THREAD

**Discussion by goldwind-ting** [rust-ndarray/ndarray#1234] [Q&A]
How to read an array form binary file?
The method is similar to `numpy.fromfile`.

**Comment by adamreichold** [comment]
If you are referring to NumPy's [NPY format](https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html), there are two crates which implement this: [ndarray-npy](https://crates.io/crates/ndarray-npy) and [npyz](https://crates.io/crates/npyz).

**Reply by goldwind-ting** [reply]
No, the file maybe a text or any other format.
![image](https://user-images.githubusercontent.com/63939636/204191699-47f77ea4-5f5a-4a77-9ab2-a62c453c8ec4.png)
